clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -820,6 +848,8 @@ private void onFailure(final long currentTimeMs,
             } else if (responseError == Errors.GROUP_AUTHORIZATION_FAILED) {
```

Comment:
Do you know why we don't handle `COORDINATOR_NOT_AVAILABLE`? I think that we should treat it like `NOT_COORDINATOR`.

Reply:
It is indeed being treated as non-retriable in the legacy coordinator (fails with a KafkaException), but I truly don't know or see any reason for it. So I made the changes here to treat it just like the NOT_COORDINATOR (try to find a new coordinator and retry), agree they should be treated similarly. 
Just for the record, in this sense we would have a different behaviour in the new consumer, on the `committed` API. The legacy one fails with `KafkaException` if it gets a COORDINATOR_NOT_AVAILABLE in the fetch response, vs the new async consumer, that would retry. I still think that should be ok given that the error is defined as retriable, and the behaviour  follows the API contract for `committed`.

Reply:
That seems to be fine.